### PR TITLE
Fix UTXO Tracking + DEX settings per network

### PR DESCRIPTION
**Pull request by jsolman**
## Changes
* Fix UTXO Tracking
* Fix withdrawing with Fees and improve withdraw reliability
* Fix bug in monitoring for transaction completed
* Move DEX script hash to be per network.
* Decrease cool down of GAS fracture and add some hysteresis.
* Also work toward allowing multiple sends of NEO or GAS from the Dashboard within the same block

### CODE REVIEW

File: src/renderer/services/dex.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const SIGNATUREREQUESTTYPE_WITHDRAWSTEP_MARK = '91';
 const SIGNATUREREQUESTTYPE_WITHDRAWSTEP_WITHDRAW = '92';
 const SIGNATUREREQUESTTYPE_CLAIM_GAS = '94';
 
-let lastUTXOWithdrawn;
+const DBG_LOG = false;
```
Comment (crishellco) [inline src/renderer/services/dex.js]: What is this? Would this be better in an environment variable that is set during the build?

File: src/renderer/services/dex.js [hunk 2]
```diff
@@ -670,7 +670,7 @@ export default {
             }
           })
           .catch((e) => {
-            // console.log(e);
+            if (DBG_LOG) console.log(e);
```
Comment (crishellco) [inline src/renderer/services/dex.js]: Does this pass the linter?

File: src/renderer/services/dex.js [hunk 3]
```diff
@@ -1276,110 +1281,110 @@ export default {
...
+        const pickedInputs = [];
+        const pickedUnspents = [];
+        let quantitySumOfPickedInputs = new BigNumber(0);
+        _.orderBy(unspents, [unspent => parseFloat(unspent.value.toString())], ['desc']).some((currentUnspent) => {
+          if (currentUnspent.reservedFor === currentWalletScriptHash) {
+            this.completeSystemAssetWithdrawals();
+            reject('Already have a UTXO reserved for your address. Completing open withdraw.');
+            return false;
+          }
+          if (currentUnspent.reservedFor && currentUnspent.reservedFor.length >= 40) {
```
Comment (crishellco) [inline src/renderer/services/dex.js]: Can you throw `40` in a `const` with a descriptive name so we know what it is?

File: src/renderer/services/dex.js (hunk 1 above)
Comment (jsolman) [reply src/renderer/services/dex.js]: Yes, we can do that later though.

File: src/renderer/services/dex.js (hunk 2 above)
Comment (jsolman) [reply src/renderer/services/dex.js]: yes.

File: src/renderer/services/dex.js (hunk 3 above)
Comment (jsolman) [reply src/renderer/services/dex.js]: Yes, in the future; it is 20 bytes in ascii hex (40 bytes). This happens to be the length of a user or contract's script hash. I'll make a constant for it as a follow up.

**Comment by jsolman** [comment]
All the logging was added to understand what was happening while I was fixing it all. It is immensely useful to be able to turn it on.

**Pull request merged**